staging/kos/cmd/controller-manager/start_functions.go

```diff
@@ -38,31 +38,29 @@ var managedControllers map[string]startFunction
 
 func init() {
 	managedControllers = map[string]startFunction{
-		"subnet-validation-contoller": startSubnetValidationController,
-		"ipam-controller":             startIPAMController,
+		"kos-subnet-validation-controller": startSubnetValidationController,
+		"kos-ipam-controller":              startIPAMController,
 	}
 }
 
 type controllerContext struct {
-	k8sClientCfg    *rest.Config
```

Comment:
Why removing the configs from the controller context? It makes for a less concise start functions signature. If it's to allow every config having a different user-agent, we can get around that by setting the user agent within the start function.

Reply:
Yes, it is to allow each controller to have a distinct user agent --- without duplicating that code in each controller.